apps/mobile/src/libs/components/categories-form/categories-form.tsx

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+
+import { Button, Checkbox, Text, View } from "~/libs/components/components";
+import { useAppForm, useCallback, useFormController } from "~/libs/hooks/hooks";
+import { toastMessage } from "~/libs/packages/toast-message/toast-message";
+import { globalStyles } from "~/libs/styles/styles";
+import {
+	type CategoryDto,
+	quizCategoriesValidationSchema,
+} from "~/packages/quiz/quiz";
+
+import { styles } from "./styles";
+
+type Properties = {
+	categories: CategoryDto[];
+};
+
+const NO_CATEGORIES_SELECTED = 0;
+
+const CategoriesForm: React.FC<Properties> = ({ categories }) => {
+	const { control, errors, handleSubmit, trigger } = useAppForm({
+		defaultValues: {
+			categories: [] as number[],
+		},
+		validationSchema: quizCategoriesValidationSchema,
+	});
+
+	const { field } = useFormController({
+		control,
+		name: "categories",
+	});
+
+	const { onChange, value: selectedCategories } = field;
+
+	const onSubmit = (): void => {
+		const selectedLabels = categories
+			.filter((category) => selectedCategories.includes(category.id))
+			.map((category) => category.name)
+			.join(", ");
+
+		if (selectedLabels.length > NO_CATEGORIES_SELECTED) {
+			toastMessage.info({ message: `Selected Categories: ${selectedLabels}` });
+		}
+	};
+
+	const handleFormSubmit = async (): Promise<void> => {
+		const isValid = await trigger("categories");
+
+		if (isValid) {
+			void handleSubmit(onSubmit)();
+		}
+	};
```

Comment:
This is all we need here
```suggestion
		const handleFormSubmit = useCallback((): void => {
		void handleSubmit(onSubmit)();
	}, [handleSubmit, onSubmit]);
```

Reply:
It’s been really helpful! Thanks